clean.sh

```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+git clean -xdf
+git submodule foreach --recursive git clean -xdf
+git submodule foreach --recursive git reset --hard
+
+exit $?
```

Comment:
`exit $?` is inaccurate since it'll only report if the last command failed - maybe use `set -e` so the script fails if any command fails?

Reply:
Fair.